app/scripts/metamask-controller.js

```diff
@@ -4529,7 +4529,11 @@ export default class MetamaskController extends EventEmitter {
             this.permissionController,
             { origin },
           ),
-
+        revokePermissionsForOrigin: (permissionKeys) => {
+          this.permissionController.revokePermissions({
+            [origin]: permissionKeys,
+          });
```

Comment:
@shanejonas and I discussed and we may want to change the method signature for the hook so we don't necessarily have to do this formatting here. But for now this works.